### DISCUSSION TITLE
Support HMIP-RGBW light entity in single channel mode

### DISCUSSION THREAD

**Discussion by plaxtika** [SukramJ/aiohomematic#1077] [Ideas]
### The problem

The light entity for the Channels of the HMIP-RGBW not showing any state (on/off dimming)

On click the light goes on and of but the state in HA is „unbekannt“

**Comment by SukramJ** [comment]
Your description is not very clear to me.
Please post some screen shots from HA and CCU.
Is it the first version of the custom component you use this device?
In which mode is the device operated?

**Comment by plaxtika** [comment]
![IMG_0567](https://github.com/danielperna84/hahomematic/assets/35133439/2d4e3ff8-c7f5-4b72-9823-fba57903da87)
![IMG_0568](https://github.com/danielperna84/hahomematic/assets/35133439/140cab1e-9614-4433-9e90-4ad9fa0575dc)
![IMG_0569](https://github.com/danielperna84/hahomematic/assets/35133439/84b5cbbb-c889-4141-bc10-af74798bb986)

**Comment by plaxtika** [comment]
I click the on button - light is on, in reality and in ccu3 but the state is it not in HA. 

Yes is the first version i use custom comp. With it. 

The device is in single channel mode.

**Comment by SukramJ** [comment]
Try this:
Execute this service (dev tools/services)`Homematic(IP) Local: Clear the cache` and restart HA.

**Comment by plaxtika** [comment]
Ok i did, no change. Same issue

**Comment by SukramJ** [comment]
I don't know the device in this mode (I don't have one either). But first I'll go on vacation and only then can I take care of it.

**Reply by plaxtika** [reply]
No Problem, have fun 👍🏻 the problem can wait ✌🏻

**Comment by SukramJ** [comment]
Bitte häng doch mal eine aktuelle [Gerätebeschreibung](https://github.com/danielperna84/custom_homematic#homematicip_localexport_device_definition) hier an. Mit den uns zur Verfügung stehenden Gerätedefinitionen sollte der PWM Modus kein Problem sein.

Um sicher zu gehen führe vor dem Export noch mal den Service (Entwicklertools/Dienste) [Clear Cache](https://github.com/danielperna84/custom_homematic#homematicip_localclear_cache) aus und starte HA neu.

Hänge bitte einen Screenshoot des Gerätes in HA an

**Comment by plaxtika** [comment]
[HmIP-RGBW.zip](https://github.com/danielperna84/hahomematic/files/11860367/HmIP-RGBW.zip)
[HmIP-RGBW_paramset.zip](https://github.com/danielperna84/hahomematic/files/11860368/HmIP-RGBW_paramset.zip)

**Reply by SukramJ** [reply]
> Hänge bitte einen Screenshoot des Gerätes in HA an

**Reply by SukramJ** [reply]
Kannst auch mal die 1.37.4 ausprobieren.

**Comment by plaxtika** [comment]
Hi,Hab ich drauf, leider keine Änderung. Alles wie bisher …-- mit freundlichen Grüßen,with best regards,Marco MännerAm 25.06.2023 um 21:03 schrieb SukramJ ***@***.***>:﻿
Kannst auch mal die 1.37.4 ausprobieren.

**Reply by SukramJ** [reply]
> Hänge bitte einen Screenshoot des Gerätes in HA an

Ohne diese Screenshots wird es nicht weiter gehen!

**Comment by plaxtika** [comment]
![image](https://github.com/danielperna84/hahomematic/assets/35133439/56664301-1bff-450e-9464-1b4358fc2380)

**Comment by plaxtika** [comment]
![image](https://github.com/danielperna84/hahomematic/assets/35133439/473a2777-701f-41a6-8f2e-dff2a5071562)

**Comment by plaxtika** [comment]
![image](https://github.com/danielperna84/hahomematic/assets/35133439/9e8d0056-6ce1-4544-b349-d04a5a68abdc)

**Comment by plaxtika** [comment]
![image](https://github.com/danielperna84/hahomematic/assets/35133439/2b887cba-7587-47a6-9804-8fe6b9ced4e9)

**Comment by plaxtika** [comment]
Passt das so ?

**Reply by SukramJ** [reply]
Ja, Danke.
Auf einem Bild ist zu sehen, das der `Value State` `restored` ist, d.h. es wurden keine Daten von der CCU empfangen.
Funktionieren die anderen Geräte? Spätestens nach einer Zustandsänderung sollte an einer Entität der Wert `Value State` `valid` stehen.

**Reply by plaxtika** [reply]
Meine anderen HMIP Geräte funktionieren. Auch dieses Geräte funktioniert wenn ich es schalte. Die Leuchte geht an / aus / dimmt. 

Nur in HASS passiert nix am Status etc. 

https://github.com/danielperna84/hahomematic/assets/35133439/47bec3b8-bd61-4453-9e03-db1006fd4aa7

**Reply by SukramJ** [reply]
Deine anderen HmIP Geräte funktionieren, UND das Attribut Value State der Entitäten hat auch immer den Wert valid?

**Reply by plaxtika** [reply]
Ja. Hier ein Beispiel
![image](https://github.com/danielperna84/hahomematic/assets/35133439/94741bef-7a14-4f12-bb7b-db2e6df50001)

**Reply by SukramJ** [reply]
OK, dann Wissen wir, das die Integration grundsätzlich funktioniert (value state == valid), nur beim HmIP-RGBW nicht.
Da neben der light entity (speziell für HmIP-RGBW) auch der Sensor Betriebsspannung (generisch) den Status unbekannt anzeigt gehe ich erst mal von dem Problem aus, das die CCU keine Events an HA sendet. Warum auch immer.

Ich würde im nächsten Schritt das Gerät in der CCU ablernen un danach wieder anlernen, vielleicht hilft das.

**Reply by plaxtika** [reply]
Ok, das probiere ich wenn ich zeit hab. Ist jetzt natürlich in der Lampe verbaut. 

Danke
